File: hello_world.cs
using System;
using System.Diagnostics;

namespace HelloWorld
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("C#: I love tornadoes");
            var processInfo = new ProcessStartInfo
            {
                UseShellExecute = false,
                FileName = "julia",
                Arguments = "hello_world.jl"
            };

            using (var process = Process.Start(processInfo))
            {
                if (process != null) process.WaitForExit();
            }
            Console.WriteLine("C#: Boom-de-ah-da!");
        }
    }
}

File: hello_world.dart
import 'dart:io';

main() {
  print('Dart: I love async programming');
  String fileExtension = '';
  if (Platform.isWindows) {
    fileExtension = '.exe';
  }
  Process.run('csc', ['-out:hello_world_cs' + fileExtension, 'hello_world.cs'])
      .then((processResult) {
    Process.start('hello_world_cs', [], mode: ProcessStartMode.inheritStdio)
        .then((process) {
      process.exitCode.then((exitCode) {
        print('Dart: Boom-de-ah-da!');
      });
    });
  });
}
